graylog2-server/src/main/java/org/graylog2/messageprocessors/MessageFilterChainProcessor.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.messageprocessors;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.google.common.collect.ComparisonChain;
+import com.google.common.collect.Ordering;
+import org.graylog2.plugin.Message;
+import org.graylog2.plugin.Messages;
+import org.graylog2.plugin.ServerStatus;
+import org.graylog2.plugin.filters.MessageFilter;
+import org.graylog2.plugin.messageprocessors.MessageProcessor;
+import org.graylog2.shared.buffers.processors.ProcessBufferProcessor;
+import org.graylog2.shared.journal.Journal;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+
+import static com.codahale.metrics.MetricRegistry.name;
+
+public class MessageFilterChainProcessor implements MessageProcessor {
+    private static final Logger LOG = LoggerFactory.getLogger(MessageFilterChainProcessor.class);
+
+    private final List<MessageFilter> filterRegistry;
```

Comment:
Well, this is actually a List, because we sort it from the set. I think this was also pre-existing code, which I haven't touched. The class was just moved.